core/state_transition.go

```diff
@@ -646,6 +641,12 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 		}, nil
 	}
 
+	if rules.IsOptimismIsthmus {
+		// Calling st.refundOperatorCost() after st.gasRemaining is updated above,
+		// so that state refunds are taken into account when calculating operator fees.
+		st.refundIsthmusOperatorCost()
+	}
+
```

Comment:
I'm wondering if we even want to move it further down into the `else` branch, so that we only refund operator fees if it's not a simulated call? It doesn't impact real execution, but makes a UX difference.

Reply:
Good point. And the closer it is to the operator add-balance part of this, the easier it is to reason about